dask_ml/preprocessing/data.py

```diff
@@ -915,3 +916,43 @@ def inverse_transform(self, X):
             X[col] = series
 
         return X
+
+
+class PolynomialFeatures(skdata.PolynomialFeatures):
+
+    __doc__ = skdata.PolynomialFeatures.__doc__
+
+    def fit(self, X, y=None):
+
+        self._transformer = skdata.PolynomialFeatures()
+        X_sample = X
+        if isinstance(X, dd.DataFrame):
+            X_sample = X._meta_nonempty
+        if isinstance(X, da.Array):
+            X_sample = np.ones((1, X.shape[1]), dtype=X.dtype)
+
+        # pandas dataframe treated by sklearn and returns np.array
+        self._transformer.fit(X_sample)
+        copy_learned_attributes(self._transformer, self)
+        return self
+
+    def transform(self, X, y=None):
+
+        if isinstance(X, da.Array):
+            XP = X.map_blocks(self._transformer.transform, dtype=X.dtype)
+        elif isinstance(X, pd.DataFrame):
+            data = X.pipe(self._transformer.transform)
+            columns = self._transformer.get_feature_names(X.columns)
+            XP = pd.DataFrame(data=data, columns=columns)
+        elif isinstance(X, dd.DataFrame):
+            meta = pd.DataFrame(
+                columns=self._transformer.get_feature_names(X.columns), dtype=float
+            )
+            data = X.map_partitions(self._transformer.transform, meta=meta)
```

Comment:
Hmm, slight issue here I think.

`_transformer.transform` is going to (AFAICT) always return an ndarray. So we have a choice

1. mimic scikit-learn, and return a dask array here (so just `XP = X.map_partitions(self._transformer.transform)`.
2. Add a keyword like `preserve_dask_dataframe=True`, to control whether we should convert the dask array back to a dask dataframe

Right now, I think that just returning a dask array is fine here.

Reply:
I'll have to think about this. In the other approach I have I didn't use meta. There map_partitions returned a dask dataframe. I'll have a deeper look into this. Maybe it's just too late ;)

Reply:
Maybe make sure you're on a recent version of dask. Returning a dask array from `DataFrame.map_partitions` is somewhat new.